### PR TITLE
Updated to ruby 2.4 and rails 4.2.8

### PR DESCRIPTION
**Pull request by rahul2240**
The aim is to update the ruby version to 2.4.
The Rails version is updated to 4.2.8 which supports ruby version 2.4
Fixes issue #1563

### CODE REVIEW

**Comment by AEtherC0r3** [comment]
@rahul2240 please address the following notes:
1. As far as I can see the changes don't include the ruby version being updated in neither the development(vagrant), nor the testing environment(travis)
2. Are all those gem updates necessary?
3. Please rebase your branch to fix the conflicts

**Comment by rahul2240** [comment]
@AEtherC0r3 SIr,
1. The ruby version is updated on both development(vagrant), and the testing(travis) environment
2. Rails update to 4.2.8 is necessary to get rid of Bignum and Fixnum warnings.
 Ruby had two visible Integer classes: Fixnum and Bignum. Ruby 2.4 unifies them into Integer. All  C  extensions which touch the Fixnum or Bignum class need to be fixed.

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ language: ruby
 cache: bundler
 rvm:
   - 2.2.3
```
Comment (shlok007) [inline .travis.yml]: @rahul2240 why do we still have this?

**Comment by shlok007** [comment]
Further changes can be tracked on #1588 .

**Pull request closed without merging**